Cosign tests began to fail on 2024-03-XX: `cosign: Error: getting Rekor public keys: updating local metadata and targets: error updating to TUF remote mirror: invalid key`

https://github.com/containerd/nerdctl/actions/runs/8397242607/job/23000193163?pr=2894


```
=== RUN   TestComposePushAndPullWithCosignVerify
    testutil.go:567: buildkitHost="unix:///run/buildkit-nerdctl-test/buildkitd.sock"
    testregistry_linux.go:49: hostIP="10.4.0.1", listenIP="0.0.0.0", listenPort=5000
    compose_run_linux_test.go:438: localhost IP="127.0.0.1"
    compose_run_linux_test.go:441: testImageRefPrefix="127.0.0.1:5000/"
    compose_run_linux_test.go:486: projectName="nerdctl-compose-test530510162"
    compose_run_linux_test.go:494: assertion failed: res.ExitCode is not exitCode: time="2024-03-22T22:59:35Z" level=info msg="Pulling image 127.0.0.1:5000/composebuild_svc0"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: Error: getting Rekor public keys: updating local metadata and targets: error updating to TUF remote mirror: invalid key"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: remote status:{"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\"mirror\": \"[https://sigstore-tuf-root.storage.googleapis.com\](https://sigstore-tuf-root.storage.googleapis.com/)","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\"metadata\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\"root.json\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"version\": 9,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"len\": 6766,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"expiration\": \"12 Sep 24 06:53 UTC\","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"error\": \"\""
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t},"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\"snapshot.json\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"version\": 132,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"len\": 2302,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"expiration\": \"09 Apr 24 16:16 UTC\","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"error\": \"\""
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t},"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\"targets.json\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"version\": 9,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"len\": 5478,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"expiration\": \"12 Sep 24 06:13 UTC\","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"error\": \"\""
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t},"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\"timestamp.json\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"version\": 170,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"len\": 721,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"expiration\": \"29 Mar 24 16:08 UTC\","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"error\": \"\""
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t}"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t}"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: }"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: main.go:74: error during command execution: getting Rekor public keys: updating local metadata and targets: error updating to TUF remote mirror: invalid key"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: remote status:{"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\"mirror\": \"[https://sigstore-tuf-root.storage.googleapis.com\](https://sigstore-tuf-root.storage.googleapis.com/)","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\"metadata\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\"root.json\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"version\": 9,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"len\": 6766,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"expiration\": \"12 Sep 24 06:53 UTC\","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"error\": \"\""
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t},"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\"snapshot.json\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"version\": 132,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"len\": 2302,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"expiration\": \"09 Apr 24 16:16 UTC\","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"error\": \"\""
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t},"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\"targets.json\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"version\": 9,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"len\": 5478,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"expiration\": \"12 Sep 24 06:13 UTC\","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"error\": \"\""
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t},"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\"timestamp.json\": {"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"version\": 170,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"len\": 721,"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"expiration\": \"29 Mar 24 16:08 UTC\","
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t\t\"error\": \"\""
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t\t}"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: \t}"
        time="2024-03-22T22:59:36Z" level=info msg="cosign: }"
        time="2024-03-22T22:59:36Z" level=fatal msg="error while pulling image 127.0.0.1:5000/composebuild_svc0: exit status 1"
        
--- FAIL: TestComposePushAndPullWithCosignVerify (3.99s)
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -288,7 +288,7 @@ WORKDIR /go/src/github.com/containerd/nerdctl
 VOLUME /tmp
 ENV CGO_ENABLED=0
 # copy cosign binary for integration test
-COPY --from=gcr.io/projectsigstore/cosign:v2.0.0@sha256:728944a9542a7235b4358c4ab2bcea855840e9d4b9594febca5c2207f5da7f38 /ko-app/cosign /usr/local/bin/cosign
+COPY --from=gcr.io/projectsigstore/cosign:v2.2.3@sha256:8fc9cad121611e8479f65f79f2e5bea58949e8a87ffac2a42cb99cf0ff079ba7 /ko-app/cosign /usr/local/bin/cosign
 # installing soci for integration test
 ARG SOCI_SNAPSHOTTER_VERSION
 RUN fname="soci-snapshotter-${SOCI_SNAPSHOTTER_VERSION}-${TARGETOS:-linux}-${TARGETARCH:-amd64}.tar.gz" && \

